rendercv/cli/commands.py

```diff
@@ -141,57 +142,90 @@ def cli_command_render(
     input_file_path: pathlib.Path = utilities.string_to_file_path(
         input_file_name
     )  # type: ignore
-    output_directory = pathlib.Path.cwd() / output_folder_name
-
-    paths: dict[
-        Literal["latex", "pdf", "markdown", "html", "png"], Optional[pathlib.Path]
-    ] = {
-        "latex": utilities.string_to_file_path(latex_path),
-        "pdf": utilities.string_to_file_path(pdf_path),
-        "markdown": utilities.string_to_file_path(markdown_path),
-        "html": utilities.string_to_file_path(html_path),
-        "png": utilities.string_to_file_path(png_path),
+
+    # dictionary for command line arguments:
+    cli_args = {
+        "use_local_latex_command": use_local_latex_command,
+        "output_folder_name": output_folder_name,
+        "latex_path": utilities.string_to_file_path(latex_path),
+        "pdf_path": utilities.string_to_file_path(pdf_path),
+        "markdown_path": utilities.string_to_file_path(markdown_path),
+        "html_path": utilities.string_to_file_path(html_path),
+        "png_path": utilities.string_to_file_path(png_path),
+        "no_markdown": no_markdown,
+        "no_html": no_html,
+        "no_png": no_png,
+    }
+
+    # Use inspect to get the default values of the arguments:
+    sig = inspect.signature(cli_command_render)
+    cli_args_default = {
+        k: v.default
+        for k, v in sig.parameters.items()
+        if v.default is not inspect.Parameter.empty
     }
 
+    # keep the current working directory:
+    working_directory = pathlib.Path.cwd()
+
     # change the current working directory to the input file's directory (because
     # the template overrides are looked up in the current working directory):
     os.chdir(input_file_path.parent)
 
     # compute the number of steps
-    # 1. read and validate the input file
+    # 0. read the input file
+    # 1. validate the input file
     # 2. generate the LaTeX file
     # 3. render the LaTeX file to a PDF
     # 4. render PNG files from the PDF
     # 5. generate the Markdown file
     # 6. render the Markdown file to a HTML (for Grammarly)
+
+    data_as_a_dict = data.read_a_yaml_file(input_file_path)
+
+    # update the data if there are extra override arguments:
+    if extra_data_model_override_argumets:
+        key_and_values = dict()
+        key_and_values = utilities.parse_render_command_override_arguments(
+            extra_data_model_override_argumets
+        )
+        data_as_a_dict = utilities.set_or_update_values(data_as_a_dict, key_and_values)
+
+    # update the data of the rendercv settings:
+    render_cv_settings = data_as_a_dict.get("rendercv_settings", dict())
+    if render_cv_settings is None:
+        render_cv_settings = dict()
+
+    # get the render options:
+    render_options = render_cv_settings.get("render", dict())
+    render_options = utilities.update_render_settings(
```

Comment:
`update_render_settings` should get two inputs: `data_as_a_dict` and `cli_args`. Then, it should return the new `data_as_a_dict`. In here, there shouldn't be any logic related to this (`inspect` package, checking if user gave settings in the YAML, etc.)